### ISSUE TITLE
'AVCaptureDevice' is only available in Mac Catalyst 14.0 or newer

### ISSUE THREAD

**Issue by appfrilans** [stripe/stripe-ios#1797]
## Summary
Project with Stripe (SPM) does not compile anymore for Mac Catalyst with XCode 12.5. It gives a lot of warnings like this:

'AVCaptureDevice' is only available in Mac Catalyst 14.0 or newer

https://www.dropbox.com/s/nhk9iy8l10jhue6/Sk%C3%A4rmavbild%202021-04-29%20kl.%2012.48.12.png?dl=0

## Code to reproduce
Integrate Stripe with a project for mac catalyst with XCode 12.5.

## iOS version
macOS 11.2.3 and macOS 11.3

## Installation method
Via SPM

## SDK version
21.4.0

**Comment by appfrilans** [comment]
Just guessing, could it be related to MinimumOSVersion which seems to be used in at least FirebaseCoreDiagnostics.

"Deprications. Don’t use the iOS MinimumOSVersion information property list key to declare the minimum release of macOS in which your app runs. Use LSMinimumSystemVersion instead. (73890473)" from XCode 12.5 release notes: https://developer.apple.com/documentation/xcode-release-notes/xcode-12_5-release-notes

**Comment by leojkwan** [comment]
Running into the same problem, though I'm using the "mc-beta-2" branch.

**Comment by csabol-stripe** [comment]
I'm looking into this now!

**Comment by EliteTechnicalCare** [comment]
Even when making the minimum 14.0, I get "No such module 'Stripe'"

So I have to temporarily downgrade versions.

**Comment by appfrilans** [comment]
Just verified, downgrading to Xcode 12.4 works.

**Comment by appfrilans** [comment]
I also sometimes (maybe first run on new computer) get this dialog when running my macOS Catalyst app:

https://www.dropbox.com/s/kxqdpvhpdkg2i82/Screenshot%202021-05-04%20at%2011.07.49.png?dl=0

**Comment by davidme-stripe** [comment]
> I also sometimes (maybe first run on new computer) get this dialog when running my macOS Catalyst app:
> 
> https://www.dropbox.com/s/kxqdpvhpdkg2i82/Screenshot%202021-05-04%20at%2011.07.49.png?dl=0

That's worrying, can you send me (davidme@stripe.com) a zipped copy of the Stripe.framework mentioned in the error? My guess is that the code signing certificate you used to sign your app was revoked, but it's possible that something has gone wrong on our end.

**Comment by appfrilans** [comment]
> > I also sometimes (maybe first run on new computer) get this dialog when running my macOS Catalyst app:
> > https://www.dropbox.com/s/kxqdpvhpdkg2i82/Screenshot%202021-05-04%20at%2011.07.49.png?dl=0
> 
> That's worrying, can you send me ([davidme@stripe.com](mailto:davidme@stripe.com)) a zipped copy of the Stripe.framework mentioned in the error? My guess is that the code signing certificate you used to sign your app was revoked, but it's possible that something has gone wrong on our end.

I've emailed you the file. Hope you that helps.

**Comment by davidme-stripe** [comment]
Thanks! It looks like this was signed with a revoked Stripe signing certificate. The signing certificate wasn't compromised, and this binary is safe. We no longer codesign binary framework releases, so updating to the latest `Stripe.xcframework` should fix this. If you'd like to remove the invalid code signature from the version you have, you can run:

```
codesign --remove-signature Stripe.xcframework
codesign --remove-signature Stripe.xcframework/ios-armv7_arm64/Stripe.framework
codesign --remove-signature Stripe.xcframework/ios-i386_x86_64-simulator/Stripe.framework
codesign --remove-signature Stripe.xcframework/ios-x86_64-maccatalyst/Stripe.framework
```

**Comment by davidme-stripe** [comment]
Please try [21.5.0](https://github.com/stripe/stripe-ios/releases/tag/21.5.0), this should be fixed.

**Comment by pmmlo** [comment]
These Mac Catalyst errors seem to have recurred in SDK version 22.5.1.

System version:
macOS 12.4
Xcode 13.4.1

Errors include:
'AVCaptureDevice' is only available in Mac Catalyst 14.0 or newer.

Can open this in a new issue.

**Comment by pmmlo** [comment]
Opened in new issue #1980
